src/core/Akka/IO/TcpManager.cs

```diff
@@ -100,5 +119,21 @@ protected override bool Receive(object message)
                                         $"If you are going to manage your connection state, you need to communicate with Tcp.Connected sender actor. " +
                                         $"See more here: https://getakka.net/articles/networking/io.html", nameof(message));
         }
+
+        
+    }
+    static class TcpActorPropsExtensions
+    {
+        public static Props WithDispatcherIfNeeded(this Props props,
+            IEnumerable<Inet.SocketOption> options)
+        {
+            if (options.FirstOrDefault(o => o is Inet.SO.WorkerDispatcher) is
+                Inet.SO.WorkerDispatcher opt)
+            {
+                return props.WithDispatcher(opt.Dispatcher);
+            }
+
+            return props;
+        }
```

Comment:
Since I can't fathom how making TCP actors would be on the Hot path, went this route to keep things simple and DSL-y